### ISSUE TITLE
Add a docker configuration to ease testing

### ISSUE THREAD

**Issue by Maxwell2022** [db-migrate/node-db-migrate#372]
I was about to create a pull request to improve error handling when parsing the configuration when I noticed that the tests are running against all database type: pg, mysql, sqlite3 and mongodb

It's not very trivial for anyone to install and configure all these databases in order to test this library. I would recommend to create a `docker-compose` file that would take care of pulling the corresponding container and start them all.

I don't really have the time to look into it now, I'll have a look later this week if it's something that you would consider benefic.

Another solution would be to mock all the database drivers

**Comment by wzrdtales** [comment]
@Maxwell2022 I agree, I put this as a feature request, I wanted to add docker anyway as I use it everywhere.

And about the mocking of database drivers:
No doesn't make sense in this case, there are currently some integration tests together with the drivers. But it is also planned that this kind of tests are going to disappear from db-migrate and move to the place where they belong now in the new structure. -> Directly in the repo of the driver being tested.

**Comment by ahmed-hritani** [comment]
Can I still work on this, first time contributor here.

**Comment by AHFoD** [comment]
Hello is this fix available? I think i am able to help

**Comment by SD007CSE** [comment]
can I help in this issue?

**Comment by camicacace** [comment]
Hi, is this issue still available?
